create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java

```diff
@@ -63,25 +68,27 @@ void shouldLogExceptionMessageWhenExceptionOccurs() {
         assertThat(appender.getMessages(), containsString(expectedMessage));
     }
 
+
     @Test
-    void shouldReturnCorrectDoiWhenEventWithS3UriThatPointsToScopusXmlWithDOi() throws IOException {
-        var scopusFile = IoUtils.stringFromResources(Path.of("2-s2.0-0000469852.xml"));
+    void shouldExtractScopusIdentifierAndPlaceItInsideAdditionalIdentifiersObject() throws IOException {
+        var scopusFile = IoUtils.stringFromResources(Path.of(SCOPUS_XML_0000469852));
         var uri = s3Driver.insertFile(UnixPath.of(randomString()), scopusFile);
         S3Event s3Event = createS3Event(uri);
-        String content = scopusHandler.handleRequest(s3Event, CONTEXT);
-        assertThat(content, is(equalTo(HARD_CODED_DOI_IN_RESOURCE_FILE)));
+        CreatePublicationRequest createPublicationRequest = scopusHandler.handleRequest(s3Event, CONTEXT);
+        assertTrue(createPublicationRequest.getAdditionalIdentifiers().contains(new AdditionalIdentifier(ScopusHandler.ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME, SCP_ID_IN_0000469852)));
+        assertEquals(1, createPublicationRequest.getAdditionalIdentifiers().toArray().length);
```

Comment:
assertThat(yourMap, IsMapContaining.hasEntry("key","value"));

Do not limit the assertion to what you think is technically possible. Write your assertion and search for a matcher in hamcrest or assertJ. Probably it exists already

Reply:
also since you are using an xml file, put the value that exists in the xml value. The test will pass if there is a value but is the wrong one.

Reply:
`contains` is a bit misleading. the arguments array (varargs) represents the contents of the list in the exact order that they should appear. So, if you put `list, contains(foo,bar)` then you expect the list to contain only two items the first being foo, and the next bar and nothing else. As a result, this check is enough and the size check is not necessary.

Suggestion:
```
 @Test
    void shouldExtractScopusIdentifierAndPlaceItInsideAdditionalIdentifiersObject() throws IOException {
        var scopusFile = IoUtils.stringFromResources(Path.of(SCOPUS_XML_0000469852));
        var uri = s3Driver.insertFile(UnixPath.of(randomString()), scopusFile);
        S3Event s3Event = createS3Event(uri);
        CreatePublicationRequest createPublicationRequest = scopusHandler.handleRequest(s3Event, CONTEXT);
        var actualAdditionalIdentifiers = createPublicationRequest.getAdditionalIdentifiers();
        var expectedIdentifier =
            new AdditionalIdentifier(ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME, SCP_ID_IN_0000469852);
        assertThat(actualAdditionalIdentifiers, contains(expectedIdentifier));
    }
```

Reply:
Missed that part where the whole set had to match with contains, I just assumed it worked the same way as list.contains in javascript. Thanks.

The reason why I didn't follow your first example was that i couldn't make isMapContaining.hasEntry compile.

Reply:
Fiksed